src/libraries/System.Text.Json/source_generation/src/ReflectionUtils/MetadataLoadContext.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis;
+
+namespace System.Reflection
+{
+    public class MetadataLoadContext
```

Comment:
Is using the same name as the existing "System.Reflection.MetadataLoadContext" intentional?

Reply:
We shouldn't be doing that. Renaming this to `System.Text.Json.SourceGeneration.Reflection.MetadataLoadContextInternal` for now.